tests/conftest.py

```diff
@@ -124,57 +125,100 @@ def clean_serveradmin(filters):
     Query(filters).delete().commit()
 
 
-def clean_aws(vm_hostname):
-    def _get_instance_status():
-        response = ec2.describe_instances(
-            Filters=[
-                {
-                    'Name': 'instance-state-code',
-                    'Values': [
-                        str(AWS_RETURN_CODES['pending']),
-                        str(AWS_RETURN_CODES['running']),
-                        str(AWS_RETURN_CODES['shutting-down']),
-                        str(AWS_RETURN_CODES['terminated']),
-                        str(AWS_RETURN_CODES['stopping']),
-                        str(AWS_RETURN_CODES['stopped']),
-                    ]
-                },
-            ],
-            InstanceIds=[obj['aws_instance_id']],
-            DryRun=False)['Reservations'][0]['Instances'][0]['State']['Code']
-
-        return int(response)
+def _get_aws_instance_status(instance_id: str) -> int:
+    ec2 = boto3.client('ec2')
+    response = ec2.describe_instances(
+        Filters=[
+            {
+                'Name': 'instance-state-code',
+                'Values': [
+                    str(AWS_RETURN_CODES['pending']),
+                    str(AWS_RETURN_CODES['running']),
+                    str(AWS_RETURN_CODES['shutting-down']),
+                    str(AWS_RETURN_CODES['terminated']),
+                    str(AWS_RETURN_CODES['stopping']),
+                    str(AWS_RETURN_CODES['stopped']),
+                ]
+            },
+        ],
+        InstanceIds=[instance_id],
+        DryRun=False)['Reservations'][0]['Instances'][0]['State']['Code']
+
+    return int(response)
 
+
+def clean_aws(vm_hostname):
     try:
         obj = Query({'hostname': vm_hostname}, ['aws_instance_id']).get()
     except DatasetError:  # No object to clean up
         return
 
+    timeout = 120
+    ec2 = boto3.client('ec2')
     if not obj['aws_instance_id']:
         return
 
-    timeout = 120
-    ec2 = boto3.client('ec2')
     try:
         ec2.stop_instances(
             InstanceIds=[obj['aws_instance_id']], DryRun=False
         )
+        for _ in range(timeout):
+            instance_status = _get_aws_instance_status(obj['aws_instance_id'])
+            if AWS_RETURN_CODES['stopped'] == instance_status:
+                break
+            sleep(1)
     except ClientError as e:
```

Comment:
this `ClientError` is too broad, especially now that we moved another API call in there. if you want to check for not running or missing instances, please use `Client.InvalidInstanceID.NotFound` and `InvalidInstanceID` or other types of errors if applicable. see https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html
all other errors are unexpected and should not be `pass`ed so we can see them

Reply:
raising the exception if its not a `InvalidInstanceID`